src/mlpack/methods/reinforcement_learning/environment/frozen_lake.hpp

```diff
@@ -0,0 +1,470 @@
+      curCol(0)
+    {
+      // Nothing to do here.
+    }
+
+    //! Copy constructor.
+    State(State const& other) :
+      height(other.height),
+      width(other.width),
+      curRow(other.curRow),
+      curCol(other.curCol)
+    {
+      // Nothing to do here.
+    }
+
+    //! Move constructor.
+    State(State&& other) :
+      height(other.height),
+      width(other.width),
+      curRow(other.curRow),
+      curCol(other.curCol)
+    {
+      // Nothing to do here.
+    }
+
+    //! Operator= copy constructor.
+    State& operator=(State const& other)
+    {
+      if (this != &other)
+      {
+        height = other.height;
+        width = other.width;
+        curRow = other.curRow;
+        curCol = other.curCol;
+      }
+      return *this;
+    }
+
+    //! Operator= move constructor.
+    State& operator=(State&& other)
+    {
+      if (this != &other)
+      {
+        height = other.height;
+        width = other.width;
+        curRow = other.curRow;
+        curCol = other.curCol;
+      }
+      return *this;
+    }
+
+    //! Get the height of the environment board.
+    size_t NumRows() const {return height;}
+
+    //! Get the width of the environment board.
+    size_t NumCols() const {return width;}
+
+    //! Get the current row-position of the player.
+    size_t CurRow() const {return curRow;}
+
+    //! Get the current column-position of the player.
+    size_t CurCol() const {return curCol;}
+
+    //! Set the current row-position of the player.
+    size_t& CurRow() {return curRow;}
+
+    //! Set the current column-position of the player.
+    size_t& CurCol() {return curCol;}
+
+    //! Get the state representation.
+    size_t GetState() const {return width * curRow + curCol;}
+
+   private:
+    //! Locally-stored height of the environment board.
+    size_t height;
+
+    //! Locally-stored width of the environment board.
+    size_t width;
+
+    //! Locally-stored current row-position of the agent.
+    size_t curRow;
+
+    //! Locally-stored current column-position of the agent.
+    size_t curCol;
+  };
+
+  /**
+   * Implementation of action of Frozen Lake.
+   */
+  class Action
+  {
+   public:
+    enum actions
+    {
+      Up, Down, Left, Right
+    };
+    // To store the action.
+    Action::actions action;
+
+    // Track the size of the action space.
+    static const size_t size = 4;
+  };
+
+  /**
+   * Construct a Frozen Lake instance using the given constants.
+   * 
+   * @param maxSteps the maximum number of steps that the agent can
+   *    make. Default to 200.
+   * @param height Height of the environment board. Default to 4.
+   * @param width Width of the environment board. Default to 4.
+   * @param platformRate The probability of platform (walkable) tile
+   *    distribution. Default to 0.8.
+   */ 
+  FrozenLake(size_t const maxSteps=200,
+             size_t const height=4,
+             size_t const width=4,
+             double const platformRate=0.8) : 
```

Comment:
```suggestion
  FrozenLake(size_t const maxSteps = 200,
             size_t const height = 4,
             size_t const width = 4,
             double const platformRate = 0.8) : 
```

Pedantic style issue, missing space between operations.

Reply:
sure! Will change!